.circleci/config.yml

```diff
@@ -169,6 +169,94 @@ jobs:
           key: coverage-web3-wrapper-{{ .Environment.CIRCLE_SHA1 }}
           paths:
             - ~/repo/packages/web3-wrapper/coverage/lcov.info
+  test-python:
```

Comment:
What is the difference between `test-python` and `test-rest-python`? We have a single package, why can't we have a single CI process?

Reply:
The reason I set it up as a two-stage thing is to enable fast failing.

Both tasks run the same set of tests.

`test-python` just calls `python` to run the tests, so uses the default python executable (via `PATH` resolution), and runs quite fast.

`test-rest-python` runs the tests via `tox`, which automatically sets up "virtual environments" to run the tests in.  Virtual environments are used in Python to run your code in a controlled environment, testing with a specific interpreter and a specific set of package dependencies.  Typically `tox` is used to ensure compatibility with multiple versions of python (and the versions of package dependencies that are available for each of those python versions), but possibilities abound:  For example, [web3.py uses it](https://github.com/ethereum/web3.py/blob/master/tox.ini) to run their tests against multiple different ethereum node implementations.

Having said all that, _at this point_, using `tox` is not adding any value for us.  (It made sense in my local environment, because I have both Python 3.6 and 3.7 installed, and we can support both.  However, CircleCI's Python container only contains 3.5 and 3.7, and we won't support 3.5 due to lack of type annotations, so right now `tox` is only using a single virtual environment, that for 3.7.)  However, I think that we will get a lot of value out of it as our codebase grows.

So, the reason I divided the two runs from each other is so that that failures would be raised quickly by `test-python`, so that we don't have to wait for all of the `tox` environment permutations to run before finding out what's broken.  But if you feel strongly that they should be combined into one run, I'm good with that too.

Reply:
Let's keep them separate but omit the `tox` run from CI for now. We can always add it back later. Thoughts?